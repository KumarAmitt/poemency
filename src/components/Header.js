import React from 'react';

const Header = () => (
  <>
    <h1>Poemency</h1>
  </>
);

export default Header;
