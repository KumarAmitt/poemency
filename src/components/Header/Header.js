import React, { useState } from 'react';
import { useDispatch } from 'react-redux';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import InputBase from '@material-ui/core/InputBase';
import { alpha, makeStyles } from '@material-ui/core/styles';
import SearchIcon from '@material-ui/icons/Search';

import { loadPoetryByTitle } from '../../store/slicers/poetryHub';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },

  title: {
    flexGrow: 1,
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: alpha(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: alpha(theme.palette.common.white, 0.25),
    },
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing(1),
      width: 'auto',
    },
  },
  searchIcon: {
    padding: theme.spacing(0, 2),
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 0),
    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: '16ch',
      '&:focus': {
        width: '20ch',
      },
    },
  },
}));

const Header = () => {
  const [title, setTitle] = useState('');

  const dispatch = useDispatch();

  const handleChange = (e) => {
    setTitle(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(loadPoetryByTitle(title));
    setTitle('');
  };

  const classes = useStyles();
  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Typography className={classes.title} variant="h6" noWrap>
            Poemency
          </Typography>
          <div className={classes.search}>
            <div className={classes.searchIcon}>
              <SearchIcon />
            </div>
            <form onSubmit={handleSubmit}>
              <InputBase
                placeholder="Search by title..."
                classes={{
                  root: classes.inputRoot,
                  input: classes.inputInput,
                }}
                inputProps={{ 'aria-label': 'search' }}
                value={title}
                onChange={handleChange}
              />
            </form>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default Header;
