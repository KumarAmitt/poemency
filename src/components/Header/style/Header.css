.app-title {
  color: #fff;
  text-decoration: none;
}
