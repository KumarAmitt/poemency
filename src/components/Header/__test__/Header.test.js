import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import configureAppStore from '../../../store/configureStore';
import Header from '../Header';

const store = configureAppStore();

describe('Header', () => {
  test('should match snapshot of Header', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
