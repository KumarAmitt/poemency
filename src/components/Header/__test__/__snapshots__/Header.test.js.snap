// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot of Header 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h4
        className="MuiTypography-root makeStyles-title-2 MuiTypography-h4 MuiTypography-noWrap"
      >
        Poemency
      </h4>
      <div
        className="makeStyles-search-3"
      >
        <div
          className="makeStyles-searchIcon-4"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="MuiInputBase-root makeStyles-inputRoot-5"
            onClick={[Function]}
          >
            <input
              aria-label="search"
              className="MuiInputBase-input makeStyles-inputInput-6"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search by title..."
              type="text"
              value=""
            />
          </div>
        </form>
      </div>
    </div>
  </header>
</div>
`;
